src/combinatorics/additive/pluennecke_ruzsa.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2022 Yaël Dillies, George Shakan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, George Shakan
+-/
+import combinatorics.double_counting
+import data.finset.pointwise
+
+/-!
+# The Plünnecke-Ruzsa inequality
+
+This file proves Ruzsa's triangle inequality, the Plünnecke-Petridis lemma, and the Plünnecke-Ruzsa
+inequality.
+
+## Main declarations
+
+* `finset.card_sub_mul_le_card_sub_mul_card_sub`: Ruzsa's triangle inequality, difference version.
+* `finset.card_add_mul_le_card_add_mul_card_add`: Ruzsa's triangle inequality, sum version.
+* `finset.pluennecke_petridis`: The Plünnecke-Petridis lemma.
+* `finset.card_smul_sub_smul_le`: The Plünnecke-Ruzsa inequality.
```

Comment:
It would be nice to have some references for these things - I don't know of a source which calls this the Plünnecke-Petridis lemma, for example